File: 2nd_sem/code_for_presentation/dip/BadPractice.cs
public class MySQLDatabase {
    public void SaveData(string data) {
        Console.WriteLine($"Зберігаю дані в MySQL: {data}");
    }
}

public class ReportService {
    private MySQLDatabase _database = new MySQLDatabase(); // Пряма залежність! Погано!

    public void GenerateReport() {
        _database.SaveData("Звіт 2024");
    }
}

File: 2nd_sem/Presentations/soLID/dip/BadPractice.kt
class MySQLDatabase {
    fun saveData(data: String) {
        println("Зберігаю дані в MySQL: $data")
    }
}

class ReportService {
    private val database = MySQLDatabase() // Пряма залежність! Погано!

    fun generateReport() {
        database.saveData("Звіт 2024")
    }
}
